RC: Version managment phase B (#1374)
redis/docs: content/operate/rc/databases/upgrade-version.md

Reviewer: @cmilesb let's also write a note somewhere on the page that it is recommended to perform the DB upgrade during off-peak hours. Maybe in addition to the note about reviewing the release notes before upgrading.

---
Title: Upgrade Redis database version
alwaysopen: false
categories:
- docs
- operate
- rc
description: Describes when you can upgrade your database to the latest available version for each plan type.
linkTitle: Upgrade database version
weight: 37
---

Database version upgrade options depend on your plan type.

## Redis Cloud Essentials

> When a new Redis version is released, all Redis Cloud Essentials databases will be upgraded to the new version after the preview period. The databases will be updated during a [maintenance window]({{< relref "/operate/rc/subscriptions/maintenance#redis-cloud-essentials" >}}).

Reviewer: emm this is not correct anymore. 
In the near future, we will also introduce an option to upgrade an Essentials DB. No need to make too many changes in this part, but maybe be more gentle and say - Redis will notify customers when a new version is available. Something like that. Anyway, we will change it pretty soon (this is the next phase of this project)

Author: ```suggestion
All new Redis Cloud Essentials databases are on the latest available version of Redis. Redis will notify Redis Cloud Essentials users when a new database version is available. 
```

## Redis Cloud Pro

> Redis Cloud Pro databases that are not on the latest stable version of Redis can be upgraded to a later version at any time.

Reviewer: Remove the word "stable".

Author: ```suggestion
Redis Cloud Pro databases that are not on the latest available version of Redis can be upgraded to a later version at any time.
```

{{< note >}}
Reverting to a previous Redis version is not supported on Redis Cloud. 

> Before updating your Redis version, we recommend that you [back up your data]({{< relref "/operate/rc/databases/back-up-data" >}}). If you need to revert back to a previous database version, you can either:

Reviewer: "upgrading", not updating.

> - [Create a new database in the same subscription]({{< relref "/operate/rc/databases/create-database/create-pro-database-existing" >}}) with your desired version and the same port number, and [import]({{< relref "/operate/rc/databases/import-data" >}}) the backup to the new database, OR

Reviewer: Please follow the section here - https://redislabs.atlassian.net/wiki/spaces/PM/pages/4069490801/PRD+Versions+Management+In+The+Cloud#Rollback-capability.
The customer must delete the old one if they want to use the same port in the same subscription.
I would have this note as a sub-section at the bottom of the page rather than in a note.

Author: It's important enough to tell the user to back up before getting into the steps. I'll move the revert stuff into its own section, but I do want to keep the note telling the user to back up their database before getting into the actual upgrade steps.

- [Create a new database]({{< relref "/operate/rc/databases/create-database/create-pro-database-existing" >}}) with your desired version, and [migrate data]({{< relref "/operate/rc/databases/migrate-databases" >}}) and connections from the old database to the new one.

Review the breaking changes for the new database version before updating: 
- [Redis 7.2]({{< relref "/operate/rc/changelog/2023/june-2023#redis-72-breaking-changes" >}}) 
- [Redis 7.4]({{< relref "/operate/rc/changelog/july-2024#redis-74-breaking-changes" >}}) 
{{< /note >}}

### Upgrade Redis Cloud Pro database

{{< note >}}

> Upgrading a single Redis Cloud Pro database is available for selected accounts and will be rolled out gradually to other accounts in the future. If you don't see **Version upgrade** in the **More actions** menu for your database and your database version is not the latest available version, you can request to upgrade all of the databases in your subscription from the [subscription page]({{< relref "/operate/rc/subscriptions/view-pro-subscription" >}}).

Author: @NoamSternRedis Do you want to keep this note? You mentioned wanting to remove it from the embed (and thus from the view-pro-subscription page), but do we want to mention the rollout here and in the changelog?

Reviewer: I think we should remove this note

{{< /note >}}

To upgrade a Redis Cloud Pro database: 

1. Choose your database from the **Databases** list to open your database page. Select **More actions > Version upgrade**.

    {{<image filename="images/rc/databases-more-actions-menu.png" alt="The More Actions menu on the Database page." width=40% >}}
    
    You can also select **More actions > Version upgrade** from the database list.

1. Select the target version from the **Select version** list.

>     {{<image filename="images/rc/database-version-upgrade.png" alt="The Redis version upgrade screen." >}}

Reviewer: can we make this image smaller? It makes it a bit hard to follow IMHO.

Author: Fixed in [58fb57d](https://github.com/redis/docs/pull/1374/commits/58fb57dd21a50bac22e61979e37ab5b70acbc00a).

>     If your database has not been backed up before, select **Go to backup** to go to the [backup settings]({{< relref "/operate/rc/databases/back-up-data" >}}).

Reviewer: It sounds like it's an obligation. Let's write 'we recommend.."

Author: ```suggestion
    If your database has not been backed up before, we recommend that you back up your database. Select **Go to backup** to go to the [backup settings]({{< relref "/operate/rc/databases/back-up-data" >}}).
```

    {{<image filename="images/rc/button-upgrade.png" alt="The upgrade button." width=100px >}}

The database will start upgrading to the selected version immediately. The upgrade may take a few minutes. 

You can continue to use the Redis Cloud console for other tasks during the upgrade.

### Manually revert upgrade

Automatically reverting to a previous Redis version is not supported on Redis Cloud.

If you [backed up your database]({{< relref "/operate/rc/databases/back-up-data" >}}) before you upgraded your version, you can:

1. [Delete your database]({{< relref "/operate/rc/databases/delete-database" >}}) without deleting your subscription.
1. [Create a new database]({{< relref "/operate/rc/databases/create-database/create-pro-database-existing" >}}) in your subscription with the following settings:
    - **Port number**: Use the same port number as the old database.
    - **Version**: Select the original version of Redis.
1. [Import the backup files]({{< relref "/operate/rc/databases/import-data" >}}) into the new database.

This allows you to connect to the database on the previous version without changing your connection details in your application.

> If you did not back up your database before upgrading:

Author: @NoamSternRedis I left a comment in the PRD about these steps.
